File: src/main/java/gargoyle/rpycg/util/GameUtil.java
package gargoyle.rpycg.util;

import java.nio.file.Files;
import java.nio.file.Path;

public final class GameUtil {
    private GameUtil() {
        throw new IllegalStateException(GameUtil.class.getName());
    }

    public static boolean isGameDirectory(final Path path) {
        return null != path &&
                Files.isDirectory(path) &&
                Files.isDirectory(path.resolve("renpy")) &&
                Files.isDirectory(path.resolve("game")) &&
                Files.isDirectory(path.resolve("lib"));
    }
}

File: app/src/main/kotlin/gargoyle/rpycg/util/GameUtil.kt
package gargoyle.rpycg.util

import java.nio.file.Path
import kotlin.io.path.isDirectory

object GameUtil {

    fun isGameDirectory(path: Path?): Boolean =
        path?.isDirectory() == true &&
                path.resolve("renpy").isDirectory() &&
                path.resolve("game").isDirectory() &&
                path.resolve("lib").isDirectory()

}
